src/FileBackedTaskManager.java

```diff
@@ -53,18 +67,22 @@ public HashMap<Integer, SubTask> deleteSubTasks() {
 
     @Override
     public Task addTask(Task newSingleTask) {
+        super.addTask(newSingleTask);
         save();
         return super.addTask(newSingleTask);
-
     }
 
     @Override
     public Epic addEpic(Epic newEpic) {
+        super.addEpic(newEpic);
+        save();
         return super.addEpic(newEpic);
     }
 
     @Override
     public SubTask addSubTask(SubTask newSubTask) {
+        super.addSubTask(newSubTask);
+        save();
         return super.addSubTask(newSubTask);
```

Comment:
```suggestion
        SubTask subtask = super.addSubTask(newSubTask);
        save();
        return subtask;
```
У тебя выполняется лишнее добавление задачи
Выше тоже поправь.